src/modules/notLogged/pages/Questionnaire.vue

```diff
@@ -18,17 +33,81 @@ export default {
   },
   components: {
     'compani-header': CompaniHeader,
+    'meta-infos': MetaInfos,
+    start: Start,
+    end: End,
   },
-  setup () {
+  setup (props) {
+    const { courseId, questionnaireId } = toRefs(props);
+    const course = ref({});
+    const questionnaire = ref({});
+    const trainee = ref({ _id: '' });
+    const btnLoading = ref(false);
+    const startCardIndex = ref(-1);
+
+    const $store = useStore();
+    const cardIndex = computed(() => $store.state.questionnaire.cardIndex);
+
+    const getCourse = async () => {
+      const fetchedCourse = await Courses.get(courseId.value);
+      course.value = fetchedCourse;
+    };
+
+    const getQuestionnaire = async () => {
+      const fetchedQuestionnaire = await Questionnaires.get(questionnaireId.value);
+      questionnaire.value = fetchedQuestionnaire;
+    };
+
+    const created = async () => {
+      await getCourse();
+      await getQuestionnaire();
+    };
+
+    created();
+
+    const updateTrainee = t => (set(trainee.value, '_id', t));
+
+    const createHistory = async () => {
+      try {
+        btnLoading.value = true;
+        const payload = {
+          course: course.value._id,
+          questionnaire: questionnaire.value._id,
+          user: trainee.value._id,
+        };
+
+        await QuestionnaireHistories.create(payload);
+        NotifyPositive('Réponse enregistrée.');
+      } catch (e) {
+        console.error(e);
+
+        if (e.response.status === 409) return NotifyNegative(e.response.data.message);
+        NotifyNegative('Erreur lors de l\'enregistrement des réponses au questionnaire.');
+      } finally {
+        btnLoading.value = false;
+      }
+    };
+
+    const endCardIndex = computed(() => (get(questionnaire.value, 'cards') ? questionnaire.value.cards.length : null));
```

Comment:
ici ce n'est pas `null` la valuer par defaut mais 0

Reply:
d'ailleurs comment as tu fais pour tester la dernière carte de ton questionnaire ? car :
- si ton questionnaire n'a pas de carte, `endCardIndex` est null donc tu n'affiche pas le composant
- si ton questionnaire a des cartes, `endCardIndex` est strictement positif du coup tu ne l'affiche pas non plus cass en cliquant sur "Suivant", cardIndex vaut 0

Reply:
En fait, le champ cards est tjs defini. J'avais ajouté ce bout de code pour ne pas avoir l'erreur `Cannot read properties of undefined (reading 'length')` a la creationn du composant. 

Mais en relisant ce n'st pas du tout optimal, j'ai modifié la maniere de definir endCardIndexx pour que ce soit plus coherent.